### ISSUE TITLE
Use OffscreenCanvas instead of `--proxy-to-worker`

### ISSUE THREAD

**Issue by janpaul123** [emscripten-core/emscripten#13652]
I was trying to figure out how to get WORKERFS to work so I can read from `File` objects directly, and for that I needed my SDL application to run in a web worker. In order to do that I came across `--proxy-to-worker`, but ran into its limitations (e.g. it crashes on `createVertexArray` because [`getExtension('OES_vertex_array_object')` is not implemented](https://github.com/kripken/webgl-worker/blob/e64b0c6910cf8bb96b03517deab2f1f0f9377fbe/webGLWorker.js#L534)).

I was then wondering if `--proxy-to-worker` should be deprecated altogether, since we can now render to a WebGL OffscreenCanvas in a WebWorker directly. I'm still trying to figure out how to do this exactly, so maybe it's implemented already? In that case it might be good to update the documentation accordingly (and perhaps add a note to `--proxy-to-worker` that you shouldn't use it, and what the alternative is?). Thanks!!

**Comment by sbc100** [comment]
I think `--proxy-to-worker` (AKA `PROXY_TO_WORKER`) has largely been replaced by `PROXY_TO_PTHREAD` these days.

We do still have support for it though and there is still some testing.   The last time this came up I think perhaps @juj still had some valid use cases for it?

**Comment by janpaul123** [comment]
Yeah, I spent some more time trying to figure out `PROXY_TO_PTHREAD`, and it looks like it hasn't been fully implemented for SDL yet, see e.g. https://github.com/emscripten-ports/SDL2/pull/127 which is still open. So I don't think we can tell people to switch over from `--proxy-to-worker` quite yet.

It still might be worth adding a note in the docs at `--proxy-to-worker`, that `PROXY_TO_PTHREAD` might be preferable whenever possible, even if `--proxy-to-worker` doesn't get outright deprecated (if that's the direction things are going in).

**Comment by juj** [comment]
>  The last time this came up I think perhaps @juj still had some valid use cases for it?

I have actually never used --proxy-to-worker for anything. When I authored --proxy-to-pthread, I thought that would obsolete --proxy-to-worker, and on several occassions I have proposed for --proxy-to-worker to be removed, though @kripken has held on to it iiuc as he has dealt with users that have used --proxy-to-worker.

**Comment by kripken** [comment]
`proxy-to-worker` also does not require pthreads. It just proxies with `postMessage` basically, and it can proxy a single-threaded app, which can be useful.

(Perhaps when pthreads support on the web is ubiquitous that will not matter though.)
